.buildkite/scripts/common.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+set -euo pipefail
+
+WORKSPACE="$(pwd)/bin"
+
+add_bin_path(){
+    mkdir -p ${WORKSPACE}
+    export PATH="${WORKSPACE}:${PATH}"
+}
+
+with_go() {
+    mkdir -p ${WORKSPACE}
+    retry 5 curl -sL -o ${WORKSPACE}/gvm "https://github.com/andrewkroh/gvm/releases/download/${SETUP_GVM_VERSION}/gvm-linux-amd64"
+    chmod +x ${WORKSPACE}/gvm
+    eval "$(gvm $(cat .go-version))"
+    go version
+    which go
+    export PATH="$(go env GOPATH):$(go env GOPATH)/bin:${PATH}"
+    echo -e "\nPATH="${PATH}"" >> dev-tools/integration/.env
+}
+
+with_docker_compose() {
+    mkdir -p ${WORKSPACE}
+    retry 5 curl -SL -o ${WORKSPACE}/docker-compose "https://github.com/docker/compose/releases/download/${DOCKER_COMPOSE_VERSION}/docker-compose-linux-x86_64"
+    chmod +x ${WORKSPACE}/docker-compose
+    docker-compose version
+    export PATH="${WORKSPACE}:${PATH}"
+    echo -e "\nPATH="${PATH}"" >> dev-tools/integration/.env
```

Comment:
I realised that this file it is committed in the repository. Usually, I try to keep the files as they are in the CI.

Could it be done this in some other way to keep this file without modifications ? @sharbuz 

Is this file being modified already when it is run in Jenkins ? 

cc @alexsapran

Reply:
After discussing this a bit, here is what we can do. 

1. Create a pre-command hook with this
```bash
....
WORKSPACE="$(pwd)/bin"
mkdir -p ${WORKSPACE}
export PATH="${PATH}:${WORKSPACE}"
....
```

2. don't do any `.env` manipulation in these scripts, so we don't need to do these (this also applies for the `with_docker`)
```
export PATH="${WORKSPACE}:${PATH}"
echo -e "\nPATH="${PATH}"" >> dev-tools/integration/.env
```

3. Remove the function `add_bin_path` and its usage.

The rationale is that we always require to do `add_bin_path` so it makes sense to be in pre-command and once it's there, the required dependencies (kind, docker-compose, docker, etc) are going to be available for usage.

Reply:
@alexsapran , I suppose, it won't work on the pre-command step, because each agent has its own PATH with the name of k8s container please take a look: https://buildkite.com/elastic/fleet-server/builds/323#0188292f-5215-402a-a295-60e819a6b823
Also, the variables which are defined in the one command of the stage (let's say one virtual bash terminal) aren't accessible in other ones.
It means we need to define and export PATH for each agent. We can use 2 ways, as we discussed earlier:
1. use file - I agree with @mrodm, we shouldn't use the current CI files, and we can use our own file for that
or
2. use build meta-data - https://buildkite.com/docs/agent/v3/cli-meta-data
In a second way, I would test it, because I didn't find info regarding the accessibility of the exported variables without the "get" command. I mean if we need to get the value every time when it's needed - it's fine, but it shouldn't be accessible everywhere without the "get" command, because in this case, we'll overwrite the variable in the other agents.